app/src/main/java/com/github/se/studybuddies/viewModels/VideoCallViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package com.github.se.studybuddies.viewModels
+
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.studybuddies.navigation.NavigationActions
+import com.github.se.studybuddies.ui.video_call.CallState
+import com.github.se.studybuddies.ui.video_call.VideoCallAction
+import com.github.se.studybuddies.ui.video_call.VideoCallState
+import io.getstream.video.android.core.Call
+import io.getstream.video.android.core.StreamVideo
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class VideoCallViewModel
+@Inject
+constructor(val uid: String, val call: Call, val navigationActions: NavigationActions) :
+    ViewModel() {
+
+  data class UiState(val isCameraEnabled: Boolean = true, val isMicrophoneEnabled: Boolean = true)
+
+  var callState by mutableStateOf(VideoCallState(call))
+    private set
+
+  fun onAction(action: VideoCallAction) {
+    when (action) {
+      is VideoCallAction.JoinCall -> joinCall()
+      is VideoCallAction.LeaveCall -> leaveCall()
+    }
+  }
+
+  val state: StateFlow<UiState> =
+      flow {
+            val isCameraEnabled = call.camera.isEnabled.value
+            val isMicrophoneEnabled = call.microphone.isEnabled.value
+            emit(
+                UiState(
+                    isCameraEnabled = isCameraEnabled, isMicrophoneEnabled = isMicrophoneEnabled))
+          }
+          .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), UiState())
```

Comment:
The number 5000 for the input stopTimeoutMillis is a fixed wait to not loose the call when small change occurs, I don't really like the concept of a fixed time as it can be too short or too long depending on situation. From what I read here : https://blog.p-y.wtf/whilesubscribed5000 and here : https://stackoverflow.com/questions/76128564/sharingstarted-whilesubscribed-vs-lazily you could just replace WhileSubscribed(5000) with Lazily in this situation.

Reply:
Right, I just left it as it was since it was given in the example of the coaches but from what I see on the threads you mentionned, I will change it to lazily since it seems to fit the use. Thanks !